src/main/scala/chisel3/iotesters/TreadleBackend.scala

```diff
@@ -5,35 +5,54 @@ package chisel3.iotesters
 import chisel3.{ChiselExecutionSuccess, Element, Mem, assert}
 import chisel3.experimental.MultiIOModule
 import chisel3.internal.InstanceId
+import chisel3.internal.firrtl.Circuit
 import firrtl.{FirrtlExecutionFailure, FirrtlExecutionSuccess, LowForm}
 import treadle.TreadleTester
 
-private[iotesters] class TreadleBackend(
-  dut: MultiIOModule,
-  firrtlIR: String,
-  optionsManager: TesterOptionsManager = new TesterOptionsManager
-)
-extends Backend(_seed = System.currentTimeMillis()) {
-
-  val treadleTester = new TreadleTester(firrtlIR, optionsManager, LowForm)
-  reset(5) // reset firrtl treadle on construction
+private[iotesters] class TreadleBackend(_seed: Long = System.currentTimeMillis()) extends Backend(_seed) {
+  private var _dut: Option[MultiIOModule] = None
+  private var _firrtlIR: Option[String] = None
+  private var _optionsManager: Option[TesterOptionsManager] = None
+  private var _interpretiveTester: Option[TreadleTester] = None
+  private var _portNames: Option[Map[Element, String]] = None
+  def dut = _dut.get
+  def interpretiveTester = _interpretiveTester.get
```

Comment:
Not sure why this became interpretiveTester seems like just created a lot name changes. I don't think it's a big deal but just curious.

Reply:
I think I was trying to consolidate firrtl-interpreter and treadle.